### PR TITLE
Task/dtserwfour 235 display currency codes beneath available funds labels

### PR DESCRIPTION
**Pull request by shanavas123**

### CODE REVIEW

File: transferui/src/main/java/com/hyperwallet/android/ui/transfer/viewmodel/CreateTransferViewModel.java [hunk 1]
```diff
@@ -112,44 +116,48 @@
 
     /**
      * Initialize Create Transfer View Model with designated transfer source token
-     *
-     * @param sourceToken              Source token that represents either from User, Bank or PrepaidCard tokens
+     *  @param sourceToken              Source token that represents either from User, Bank or PrepaidCard tokens
      * @param transferRepository       Transfer repository for making transfer calls to Hyperwallet
      * @param transferMethodRepository Transfer Method repository for making transfer method calls to
-     *                                 Hyperwallet
+ *                                 Hyperwallet
```
Comment (rbanikPP) [inline transferui/src/main/java/com/hyperwallet/android/ui/transfer/viewmodel/CreateTransferViewModel.java:122]: Please correct indentation

File: transferui/src/main/java/com/hyperwallet/android/ui/transfer/view/CreateTransferFragment.java [hunk 2]
```diff
@@ -87,7 +87,7 @@
     private static final String NOTES_TAG = "NOTES_TAGGED";
     private static final String ELLIPSIS = "...";
     private static final int NOTES_MAX_LINE_LENGTH = 40;
-    private static final String REGEX_ONLY_NUMBER = "[^0-9]";
+    public static final String REGEX_ONLY_NUMBER = "[^0-9]";
```
[the remark this answers is not in the document]
Comment (shanavas123) [reply transferui/src/main/java/com/hyperwallet/android/ui/transfer/view/CreateTransferFragment.java:90]: we need to access from viewModel

File: transferui/build.gradle [hunk 3]
```diff
@@ -15,6 +15,7 @@ dependencies {
     implementation project(":transfermethodrepository")
     implementation project(":userrepository")
     implementation project(":commonui")
+    implementation project(":userbalancerepository")
```
Comment (vwagh-hw) [inline transferui/build.gradle:18]: Why are we creating a new repository module? It should be part of `userrepository` module
Similar to how `receiptrepository` module has repository for `PrepaidCard` and `User`
Create `UserBalanceRepository` inside `com.hyperwallet.android.ui.user.repository`

I will review rest of the PRs after this is fixed

File: settings.gradle [hunk 4]
```diff
@@ -7,4 +7,5 @@ include ':commonui',
         ':transferrepository',
         ':transferui',
         ':userrepository',
+        ':userbalancerepository',
```
Comment (vwagh-hw) [inline settings.gradle:10]: How will you get prepaid card balance?
Look at iOS PR, he is creating  `BalanceRepository` which then has `PrepaidCardBalanceRepository` and `UserBalanceRepository`

Comment (shanavas123) [reply settings.gradle:10]: prepaid card balance in transfermethodrepository 
Screenshot 2021-03-10 at 12 42 20 PM

refactor prepaid card balance transfermethodrepository  to balance repository ?

Comment (shanavas123) [reply settings.gradle]: added PrepaidCardBalanceRepository in UI - SDK

File: balancerepository/build.gradle [hunk 5]
```diff
@@ -0,0 +1,12 @@
+apply from: "$rootProject.projectDir/android-library.gradle"
+description = 'Hyperwallet User Balance Repository SDK for Android to integrate with the Hyperwallet Platform'
```
[the remark this answers is not in the document]
Comment (shanavas123) [reply balancerepository/build.gradle:2]: updated description as Hyperwallet Balance Repository SDK...

File: balancerepository/build.gradle [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+project.ext {
+    mavenName = 'Hyperwallet Android User Balance Repository SDK'
```
[the remark this answers is not in the document]
Comment (shanavas123) [reply balancerepository/build.gradle:4]: updated mavenName as Hyperwallet Android Balance Repository SDK

File: balancerepository/src/main/java/com/hyperwallet/android/ui/balance/repository/PrepaidCardBalanceRepository.java [hunk 7]
```diff
@@ -0,0 +1,30 @@
+package com.hyperwallet.android.ui.balance.repository;
```
[the remark this answers is not in the document]
Comment (shanavas123) [reply balancerepository/src/main/java/com/hyperwallet/android/ui/balance/repository/PrepaidCardBalanceRepository.java:1]: copyright added

File: balancerepository/src/main/java/com/hyperwallet/android/ui/balance/repository/PrepaidCardBalanceRepositoryFactory.java [hunk 8]
```diff
@@ -0,0 +1,26 @@
+package com.hyperwallet.android.ui.balance.repository;
```
[the remark this answers is not in the document]
Comment (shanavas123) [reply balancerepository/src/main/java/com/hyperwallet/android/ui/balance/repository/PrepaidCardBalanceRepositoryFactory.java:1]: copyright added

File: transferui/src/main/java/com/hyperwallet/android/ui/transfer/viewmodel/CreateTransferViewModel.java [hunk 9]
```diff
@@ -119,37 +123,42 @@
...
 
         // initialize
         initialize();
     }
 
     /**
      * Initialize Create Transfer View Model with designated transfer source token
-     *
-     * @param transferRepository       Transfer repository for making transfer calls to Hyperwallet
+     *  @param transferRepository       Transfer repository for making transfer calls to Hyperwallet
```
[the remark this answers is not in the document]
Comment (shanavas123) [reply transferui/src/main/java/com/hyperwallet/android/ui/transfer/viewmodel/CreateTransferViewModel.java:147]: fixed

File: transferui/src/main/java/com/hyperwallet/android/ui/transfer/viewmodel/CreateTransferViewModel.java [hunk 10]
```diff
@@ -646,6 +679,20 @@ private boolean hasTransferAmountChanged(@Nullable final Transfer transfer) {
                 transfer.getDestinationAmount(), mTransferAmount.getValue());
     }
 
+    private String getCurrencyCode(List<Balance> balances)
```
[the remark this answers is not in the document]
Comment (shanavas123) [reply transferui/src/main/java/com/hyperwallet/android/ui/transfer/viewmodel/CreateTransferViewModel.java:682]: updated as getCurrencyCodes

File: transferui/src/main/java/com/hyperwallet/android/ui/transfer/viewmodel/CreateTransferViewModel.java [hunk 11]
```diff
@@ -646,6 +679,20 @@ private boolean hasTransferAmountChanged(@Nullable final Transfer transfer) {
...
+        StringBuilder builder = new StringBuilder();
+        for (Balance balance:balances) {
+            if (Double.parseDouble(balance.getAmount().replaceAll(REGEX_ONLY_NUMBER, EMPTY_STRING)) != 0.0) {
+                if (builder.length() > 0){
+                    builder.append(COMMA_SEPARATOR);
+                }
+                builder.append(balance.getCurrency());
+            }
+        }
+        return builder.toString();
```
[the remark this answers is not in the document]
Comment (shanavas123) [reply transferui/src/main/java/com/hyperwallet/android/ui/transfer/viewmodel/CreateTransferViewModel.java:693]: updated logic with TextUtils.join

File: balancerepository/src/main/java/com/hyperwallet/android/ui/balance/repository/PrepaidCardBalanceRepositoryImpl.java [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+import com.hyperwallet.android.model.balance.Balance;
+import com.hyperwallet.android.model.balance.PrepaidCardBalanceQueryParam;
+import com.hyperwallet.android.model.paging.PageList;
+
+import java.util.ArrayList;
+
+public class PrepaidCardBalanceRepositoryImpl implements PrepaidCardBalanceRepository {
+
+    private Handler mHandler = new Handler();
+    private int LIMIT = 100;
```
[the remark this answers is not in the document]
Comment (shanavas123) [reply balancerepository/src/main/java/com/hyperwallet/android/ui/balance/repository/PrepaidCardBalanceRepositoryImpl.java:38]: added

File: balancerepository/src/main/java/com/hyperwallet/android/ui/balance/repository/UserBalanceRepositoryImpl.java [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+import com.hyperwallet.android.model.balance.Balance;
+import com.hyperwallet.android.model.balance.BalanceQueryParam;
+import com.hyperwallet.android.model.paging.PageList;
+
+import java.util.ArrayList;
+
+public class UserBalanceRepositoryImpl implements UserBalanceRepository {
+
+    private Handler mHandler = new Handler();
+    private int LIMIT = 100;
```
[the remark this answers is not in the document]
Comment (shanavas123) [reply balancerepository/src/main/java/com/hyperwallet/android/ui/balance/repository/UserBalanceRepositoryImpl.java:39]: added

**Review by malaw-hw — APPROVED** [review]

**Review by vwagh-hw — APPROVED** [review]

**Review by rbanikPP — APPROVED** [review]

**Pull request merged**